core/src/plugins/filed/python/libcloud/BareosLibcloudApi.py

```diff
@@ -101,79 +131,142 @@ def __init__(self, options, last_run, tmp_dir_path):
             BareosLibcloudApi.redirect_stdout_stderr()
         except Exception as e:
             jobmessage(
-                M_ERROR, "Could not redirect stdout and/or stderr: %s" % (str(e))
+                bareosfd.M_ERROR,
+                f"Could not redirect stdout and/or stderr: {str(e)}",
             )
             raise
 
-        jobmessage(M_INFO, "Starting BucketExplorer")
+        jobmessage(bareosfd.M_INFO, "Starting BucketExplorer")
         try:
             self.bucket_explorer.start()
         except Exception as e:
-            jobmessage(M_ERROR, "Start BucketExplorer failed: %s" % (str(e)))
+            jobmessage(bareosfd.M_ERROR, f"Start BucketExplorer failed: {str(e)}")
             raise
 
-        jobmessage(M_INFO, "Starting Workers")
+        jobmessage(bareosfd.M_INFO, "Starting Workers")
         try:
             for w in self.worker:
                 w.start()
         except Exception as e:
-            jobmessage(M_ERROR, "Start Worker failed: %s" % (str(e)))
+            jobmessage(bareosfd.M_ERROR, f"Start Worker failed: {str(e)}")
             raise
 
     def worker_ready(self):
+        """
+        Return True if all workers are ready.
+        """
+        debugmessage(
+            120,
+            f"worker_ready(): count_worker_ready: {self.count_worker_ready}, "
+            f"number_of_worker: {self.number_of_worker}",
+        )
         return self.count_worker_ready == self.number_of_worker
 
     def bucket_explorer_ready(self):
+        """
+        Return True if the bucket explorer is ready.
+        """
         return self.count_bucket_explorer_ready == 1
 
     def check_worker_messages(self):
+        """
+        Process and handle messages from workers and the bucket explorer.
+        Returns WorkerState.SUCCESS or WorkerState.ABORT.
+        """
         while not self.message_queue.empty():
+            self.last_worker_message_time = int(time.time())
```

Comment:
I wonder if that int resolution is enough, or if we should goes for a 
time.time_ns() ?
(valid for the whole code, but time_ns() was introduced in python 3.7, in postgresql plugin we have this defined https://github.com/bareos/bareos/blob/03031eeebbc96a3b630c111f6c6bdcb823822e1c/core/src/plugins/filed/python/postgresql/bareos-fd-postgresql.py#L89